libyear-gradle-plugin/src/main/kotlin/com/libyear/LibYearExtension.kt

```diff
@@ -20,6 +20,8 @@ open class LibYearExtension {
 
   var validator: DependencyValidatorSpec = defaultValidator
 
+  var maxTransitiveDepth: Int = 100
```

Comment:
I think a default value of 100 for the depth parameter is surprising.

From the users perspective, the default behavior should be one of the extremes: "report first level dependencies only", or "by default, the report is exhaustive (= all dependencies from the entire graph)".

I suggest the following:

- negative values like -1 disable the limit reliably
- 0 and positive numbers place an upper bound to the depth of the dependency traversal
- we set the default value of the depth parameter to zero from within `LibYearExtension`

I think a report of the first-level dependencies is (A) quite useful for a first approximation, and (B) cheap in comparison to the full report.
Users of the plugin can then consciously decide to go for the full report, even if the report will take much longer to complete, to get the big picture.

Would you agree?

Reply:
Agree, 100 is arbitrary and there may very well be 100 node-trees out there more often than I think.
I'll use -1 or null for no limit!

I'm down to set the default for `LibYearExtension` to 0. That's was what I was expecting when I first installed it.

Reply:
> That's was what I was expecting when I first installed it.

Perfect 😄  Thank you